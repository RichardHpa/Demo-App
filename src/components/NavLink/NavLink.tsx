import { NavLink as RouterNavLink } from 'react-router-dom'
import clsx from 'clsx'

import type { FC, ReactNode } from 'react'

interface NavLinkProps {
  to: string
  className?: string
  children: ReactNode
}

export const NavLink: FC<NavLinkProps> = ({ to, children, className }) => {
  return (
    <RouterNavLink
      to={to}
      className={clsx(
        'text-gray-700 dark:text-gray-200 hover:text-gray-900 dark:hover:text-gray-100 px-3 py-2 rounded-md text-sm font-medium hover:underline',
        className,
      )}
    >
      {children}
    </RouterNavLink>
  )
}
