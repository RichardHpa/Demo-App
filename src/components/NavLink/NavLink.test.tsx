import { setMatchMediaMock } from 'tests/matchMediaMock'
import { render, screen } from 'tests/renderHelpers'

import { NavLink } from './NavLink'

describe('NavLink', () => {
  beforeEach(() => {
    setMatchMediaMock()
  })
  test('renders NavLink', () => {
    render(<NavLink to='/'>Test</NavLink>)

    const link = screen.getByRole('link', { name: /test/i })

    expect(link).toBeInTheDocument()

    expect(link).toMatchSnapshot()
  })

  test('renders NavLink with className', () => {
    render(
      <NavLink to='/' className='test'>
        Test
      </NavLink>,
    )

    const link = screen.getByRole('link', { name: /test/i })

    expect(link).toHaveClass('test')
  })
})
