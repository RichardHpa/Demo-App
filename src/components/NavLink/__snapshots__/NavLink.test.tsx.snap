// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavLink renders NavLink snapshot 1`] = `
<div>
  <a
    aria-current="page"
    class="text-gray-700 dark:text-gray-200 hover:text-gray-900 dark:hover:text-gray-100 px-3 py-2 rounded-md text-sm font-medium hover:underline active"
    href="/"
  >
    Test
  </a>
</div>
`;
